### PR TITLE
File new behavior

### LINKED ISSUE #107: New source function
**Problem**
Currently, entering source in the first session's blank editor presents no problem, until you go to save it, which presents an "EISDIR, illegal operation on a directory '/new-project' error.

**Solution**
The common workflow in BSE and PT does not require a user to create (and decide on a name for) a new source code file before entering source code in the editor (although it's an option they can choose to do).
1. Change New File function to simply clear the editor (do not display the filename dialog) and assume a default filename of "untitled#" where "#" is a numeric value starting at 1 (for each new session) and increasing for each selection of New File.  NOTE: Assume the file will be saved in the current project folder, but do not force it to be.  It's best if it's created/treated in a special way that allows it to remain easily accessible to the user (during this session or the next) but for the system to know that it has never really been "saved" by the user.
2. Selecting the Save File function now behaves like a Save As function; it displays the filename dialog (that used to be displayed by the New File function) and is pre-populated with the default filename "untitled#" with the entire filename selected so that any keyboard entry replaces it.
   - Note: Or the default filename can appear in the field but grayed-out and disappears upon the entry of the first character from the keyboard.

Add Ctrl-N as the key shortcut that activates the New source feature.

### PR DESCRIPTION
**Pull request by spatializes**
#### What's this PR do?

Changes logic for new files and saving files. New files are now created as blank pages and named with untitled# (with # being the next number after the highest numbered untitled file in your working directory).

Unsaved files will Save As, prompting a Save As overlay with Untitled# showing in the filename field. 
Saved files will simply save to their current name.

Ctrl-S and the save button should work for both of these circumstances.
#### Where should the reviewer start?

npm install (to get Frylord updates)
npm run build
#### How should this be manually tested?

Open ChromeIDE, create a new file (with the New File button and with Ctrl-N). Write some text into the file. You shouldn't see any errors in the Dev Tools console. Save (using the Save button or Ctrl-S), should open the Save As overlay. Untitled# (#, based on file directory) should show in the filename input. Save the untitled file. Create and save a few more untitled files. Be sure the numbers are incrementing properly. Click the Save button or press Ctrl-S on a file that is already saved. It should simply save without prompting the Save As dialog.

Create a new file. Try to click away to a different file in your directory. ChromeIDE should ask you to Save As.

Try different scenarios with opening new files, saving, and navigating to be certain everything is working as expected.
#### What are the relevant tickets?

Closes #107

### CODE REVIEW

File: plugins/editor/key-extension.js [hunk 1]
```diff
@@ -50,13 +51,35 @@ const keyExtension = {
...
+          return ((ctrlKey === true || metaKey === true) && keyCode === 78);
+        }
+    };
+
     function setCodeMirrorCommands() {
       for (let cmd in cmCommands) {
         const code = cmCommands[cmd].code;
-        cmCommands[cmd].removeCode = app.keypress(app.keypress[code], cmCommands[cmd].exec);
+        const predicate = app.keypress[code] || customPredicates[code];
+        console.log(code);
```
Comment (phated) [inline plugins/editor/key-extension.js]: leftover console.log

**Comment by PropGit** [comment]
Excellent descriptions @drquinn, thank you!  Looking forward to trying out these features.

File: src/stores/file.js [hunk 2]
```diff
@@ -1,19 +1,26 @@
 'use strict';
 
 const alt = require('../alt');
+const _ = require('lodash');
```
Comment (phated) [inline src/stores/file.js:3]: should be defined above the relative requires

**Comment by phated** [comment]
reassigned to @2fast2fourier to review since I made changes

**Pull request merged**